rfcs/2021-19-02-units-for-typography-and-layout.md

```diff
@@ -0,0 +1,603 @@
+
+
+## How will we store and use it?
+<img alt="storing and getting tokens" src="./images/unit-storing-and-getting.png" width="1200px">
+
+* While storing we can compute the value in `px/dp` and store it in a unitless way. While rendering we can attach the units.
+   ```js
+   const space = {
+      1: 8,
+      2: 12,
+      3: 16,
+   }
+   const fontSize = {
+      1: 14,
+      2: 18,
+      3: 24,
+   }
+   ```
+* With the above approach we can keep the vocabulary consistent that everything is stored in `px` for web and `dp` for react native  but while rendering the respective platforms(web/react-native) attaches the unit for the target rendering engine.
+* We will implement a generic funtion that will attach the units to the tokens.
+* For typography the units that'll be constructed will be relative(`rems` for web and `autoScale` value on text enabled for react-native apps).
+* For layout i.e height, width, padding, margin the units that'll be constructed will be absolute i.e `px`(pixels) for web and `dp` for react-native apps but with device pixel ratio to ensure consistency with various screen densities.
+# Drawbacks/Constraints
+- Vocabulary is the biggest drawback. Thinking and visualising in relative units is difficult compared to absolute units like pixels(`px`) or density independent pixels(`dp`) but we can work it out with the help of tools. We can think and store in pixels but render in relative units of the target platform(`rems` for web and `dp` with device pixel ratio for react-native apps).
+
+# Alternatives
+- Here's the [absolute unit vs relative unit matrix](#absolute-unit-vs-relative-unit-matrix) which states all the altternatives with the details.
+
+# Adoption strategy
+- **For developers** - All these decisions shouldn't matter, as this will be stored in our tokens as an abstraction. The end consumers will just use the typography and layout scale tokens for their purpose and not the actual values.
+- Migration of existing projects is out of the scope of this RFC since that will be a breaking change.
+- **For Designers** - All the designs shall be created `@1x` and the assets(images) shall be exported `@2x`, `@3x` accordingly. The standard artboard size for mobile is `360x640` for android and `375x677` for iPhone.
+- **For Designers** - When doing visual QA of any design which is implemented the comparision cannot be done by keeping the screenshot from device side by side with the design implemented `@1x`. That'll always give wrong comparision results.
+
+- We should have a different scale for each platform (mainly web & mobile) we are designing for instead of having a singular scale that accommodate all devices.
+
+# Open Questions
+1. Browser zoom and browser font size behave differently? Couldn't figure out what does browser zoom does differently like what css props does it changes so in order to have parity we need to mimic it?
```

Comment:
Zoom just seems to "stretch" the actual hardware pixels. [Ref](https://stackoverflow.com/questions/29390155/what-exactly-changes-in-the-css-rendering-when-desktop-browsers-zoom-in-or-out)

Reply:
Thanks for this one 🙌